src/modulefinder/sentry_modulefinder_linux.c

```diff
@@ -41,6 +43,53 @@ static sentry_value_t g_modules = { 0 };
 
 static sentry_slice_t LINUX_GATE = { "linux-gate.so", 13 };
 
+bool
+sentry__mmap_file(sentry_mmap_t *rv, const char *path)
+{
+    rv->fd = open(path, O_RDONLY);
```

Comment:
you don't need to keep the fd in the sentry_mmap_t struct, part of the point of mmap is that after you called mmap you can close the fd and the mmap stays valid while you don't consume filedescriptors.  So you should store the fd on the stack and close it at the end.